File: src/java/magma/app/io/Platform.java
package magma.app.io;

public enum Platform {
    TypeScript("node", "ts"),
    Magma("magma", "mgs"),
    Windows("windows", "h", "c");

    public final String root;
    public final String[] extension;

    Platform(final String root, final String... extensions) {
        this.root = root;
        this.extension = extensions;
    }
}

File: src/node/magma/app/io/Platform.ts
export class Platform {
	static TypeScript/*auto*/: Platform = new Platform("node", "ts");
	static Magma/*auto*/: Platform = new Platform("magma", "mgs");
	static Windows/*auto*/: Platform = new Platform("windows", "h", "c");
	root: string;
	extension: string[];
	constructor (root: string, ...extensions: string[]) {
		this/*auto*/.root = root/*auto*/;
		this/*auto*/.extension = extensions/*auto*/;
	}
}
